File: competitive_programming/codeforces/07_typescript/785a_anton_and_polyhedrons.ts
const FACES: Record<string, number> = { Tetrahedron: 4, Cube: 6, Octahedron: 8, Dodecahedron: 12, Icosahedron: 20 };

function antonAndPolyhedrons(names: string[]): number {
  return names.reduce((acc, name) => acc + FACES[name], 0);
}

if (antonAndPolyhedrons(["Icosahedron", "Cube", "Tetrahedron"]) !== 30) throw new Error("polyhedrons failed");
console.log("785A anton and polyhedrons ok");

File: competitive_programming/codeforces/01_python/785a_anton_and_polyhedrons.py
"""Codeforces 785A - Anton and Polyhedrons."""
FACES = {
    "Tetrahedron": 4,
    "Cube": 6,
    "Octahedron": 8,
    "Dodecahedron": 12,
    "Icosahedron": 20,
}


def anton_and_polyhedrons(names):
    return sum(FACES[name] for name in names)


if __name__ == "__main__":
    assert anton_and_polyhedrons(["Icosahedron", "Cube", "Tetrahedron"]) == 30
    print("785A anton and polyhedrons ok")
